### PR TITLE
replace dynamic informer and client

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/open-cluster-management/backlog/issues/18146

Refering to the investigation in https://github.com/stolostron/backlog/issues/18146#issuecomment-1001837702, replace dynamic informer to typed informer to reduce the memory usage.

Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: pkg/collectors/builder_test.go [hunk 1]
```diff
@@ -363,7 +365,8 @@ func TestBuilder_buildManagedClusterCollectorWithClient(t *testing.T) {
 		whiteBlackList    whiteBlackLister
 	}
 	type args struct {
-		client dynamic.Interface
+		client        dynamic.Interface
```
Comment (qiujian16) [inline pkg/collectors/builder_test.go:368]: do we still need dynamic client?

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

File: pkg/collectors/builder_test.go (hunk 1 above)
Comment (haoqing0110) [reply pkg/collectors/builder_test.go:368]: It's used by getHubClusterID https://github.com/stolostron/clusterlifecycle-state-metrics/blob/main/pkg/collectors/utils.go#L34, let me create another PR to remove it.

Comment (haoqing0110) [reply pkg/collectors/builder_test.go]: replace dynamic.Interface with openshift client in [commit](https://github.com/stolostron/clusterlifecycle-state-metrics/pull/67/commits/014d10ac5e489f918fa670b90417ca776b2abebb) . This PR is blocked by renaming to stolostron, will merge it once rename is done.

**Pull request merged**